### DISCUSSION TITLE
Sharing the Global Object between App Router and Pages?

### DISCUSSION THREAD

**Discussion by somersby10ml** [vercel/next.js#49798] [Help]
### Summary

Hello. I apologize if my English is not perfect.
I'm using next 13.4.2 App router and pages.

---

What I want to achieve is to use both a client-side WebSocket client and a server-side WebSocket server in the backend. 

However, because the WebSocket server requires the ws protocol, I would like to use socket.io. To install socket.io, I need to use pages (res.socket.server).

I want to install it in a global variable so that it can be accessed from anywhere. (server and client)

---
Source code description)

1. call `http://localhost:3000/api/init` from instrumentation.ts.

2. initialize global.store in `pages\api\init.ts`.
(pages\api\init.ts)

3. Store exists in `http://localhost:3000/api/get`.
(pages\api\get.ts)

4. store does not exist for `http://localhost:3000/api/test`.
(src\app\api\test\route.ts)

---
## Problem
The problem I'm facing is that it seems the global variable is not shared between `pages/api` and `app/api`, even though both are executed on the server-side.

It's possible that I have misunderstood something. How can I make the variable accessible from anywhere? It seems that the variable gets reset when a new page is requested.

In the mentioned source code, when I access /api/get (pages), the variable remains intact. However, when I access /api/test (app), {} is outputted.

---
Using global objects may not be the most common way to do this. 

How do I manage objects (like classes) globally?

Are there other ways to manage data globally besides creating a new backend (like express.js)?

I hope this accurately conveys your message. If you have any further questions or need clarification, please let me know.
[my-app.zip](https://github.com/vercel/next.js/files/11475082/my-app.zip)

### Additional information

```js
Operating System:
      Platform: win32
      Arch: x64
      Version: Windows 10 Home
    Binaries:
      Node: 18.16.0
      npm: N/A
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 13.4.2
      eslint-config-next: 13.4.2
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.0.4
```

**Comment by zoominGiladSher** [comment]
Would very much appreciate an answer to this question as I'm facing the same problem with `prom-client` being assigned to the `global` object but for some reason the `global` object isn't the same between API routes that are registered in the pages router and the app router.
